cmd/image-loader/main.go

```diff
@@ -65,34 +68,42 @@ var (
 )
 
 type opts struct {
-	versionsFile  string
-	versionFilter string
-	registry      string
-	dryRun        bool
-	addonsPath    string
+	configurationFile string
+	versionsFile      string
+	versionFilter     string
+	registry          string
+	dryRun            bool
+	addonsPath        string
 }
 
 func main() {
-	klog.InitFlags(nil)
+	var err error
 
 	logOpts := kubermaticlog.NewDefaultOptions()
+	logOpts.Format = kubermaticlog.FormatConsole
 	logOpts.AddFlags(flag.CommandLine)
 
 	o := opts{}
-	flag.StringVar(&o.versionsFile, "versions", "charts/kubermatic/static/master/versions.yaml", "The versions.yaml file path")
+	flag.StringVar(&o.configurationFile, "configuration-file", "", "Path to the KubermaticConfiguration YAML file")
+	flag.StringVar(&o.versionsFile, "versions-file", "", "The versions.yaml file path (deprecated, EE-only, used only if no -configuration-file is given)")
 	flag.StringVar(&o.versionFilter, "version-filter", "", "Version constraint which can be used to filter for specific versions")
-	flag.StringVar(&o.registry, "registry", "registry.corp.local", "Address of the registry to push to")
+	flag.StringVar(&o.registry, "registry", "", "Address of the registry to push to, for example localhost:5000")
 	flag.BoolVar(&o.dryRun, "dry-run", false, "Only print the names of found images")
 	flag.StringVar(&o.addonsPath, "addons-path", "", "Path to the folder containing the addons")
 	flag.Parse()
 
-	log := kubermaticlog.New(logOpts.Debug, logOpts.Format)
+	rawLog := kubermaticlog.New(logOpts.Debug, logOpts.Format)
+	log := rawLog.Sugar()
 	defer func() {
 		if err := log.Sync(); err != nil {
 			fmt.Println(err)
 		}
 	}()
 
+	if (o.configurationFile == "") != (o.versionsFile == "") {
```

Comment:
This is an XOR, it's not at all what we need, it fails if one of the two is provided. Anyway, after better thinking, we should probably allow when both are not specified as we have default versions. We should just fail when both are specified:
```
if o.configurationFile != "" && o.versionsFile != "" {
```

Reply:
That's not to say that this new check is maybe buggy and does represent the logic I wanted to have (and had, before adding this additional check).

Reply:
Do you prefer something like `-configuration-file /dev/null`? I think that the tool would be easier to use if when versions are not overridden both arguments can be left empty. Anyway, do as you prefer. Just fix the logic ;-)

Reply:
Yes, I prefer to have this in our CI to having a bad CLI for everyone. I agree that it's super akward, but also super clever and I was even surprised that io.ReadFile(/dev/null) works and doesn't block infinitely.

The alternative would have been to create a dummy, empty config file and specify that in our CI scripts. That wasn't worth it when /dev/null works so nicely. ^^

Once the old Helm chart is gone and we don't need to support the versions.yaml anymore, _then_ we can make the config optional and use the inbuilt one. For now having an explicit config made the logic inside more obvious.

Reply:
> Yes, I prefer to have this in our CI to having a bad CLI for everyone

I think your definition of "bad CLI" is largely debatable, but as I said, do as you like.